### PR TITLE
CAPV: Bump tests for k8s v1.31.0

### PR DESCRIPTION
**Pull request by chrischdi**
/assign @sbueringer 

Requirements:

- [ ] v1.31.0 release

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-prowjob-gen.yaml [hunk 1]
```diff
@@ -9,16 +9,16 @@ prow_ignored:
   # prowjob configuration files (presubmits, periodics)
   branches:
     main:
-      testImage: "gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240803-cf1183f2db-1.30"
+      testImage: "gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240803-cf1183f2db-1.31"
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-prowjob-gen.yaml:12]: Should we keep this image the same until we bump CR (& thus the corresponding k8s.io/* dependencies)

File: config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-prowjob-gen.yaml [hunk 2]
```diff
@@ -9,16 +9,16 @@ prow_ignored:
...
       # interval for coverage job
       interval: "24h"
       upgrades:
-      - from: "1.28"
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-prowjob-gen.yaml:16]: Let's keep this one. I think the idea was to gradually expand the tested Kubernetes versions in CAPV

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

For everyone who's wondering. We're currently updating prowjob-gen as usual. But we are not generating the tests which would require the VMC environment, because we cannot run them on upstream Prow anymore.

We will generate these jobs again once we have the new VMware environment integrated in upstream Prow

**Comment by sbueringer** [comment]
Nice catch with the 1.26 image!

We decided we'll keep the release-1.7 jobs for now. It's unclear at the moment how long we have to support 1.7.x

/lgtm

**Comment by sbueringer** [comment]
> Nice catch with the 1.26 image!

Nice catch realising before merge that it actually wasn't a nice catch :D

**Pull request merged**